sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -88,37 +100,39 @@ export class ApplicationController extends BaseController {
       );
     }
 
+    const applicationToSubmit =
+      await this.applicationService.getApplicationByIdAndUser(
+        applicationId,
+        userToken.userId,
+      );
+    if (!applicationToSubmit) {
+      throw new NotFoundException("Student Application not found.");
```

Comment:
We use this logic of throwing a NotFoundException when an object is undefined. Can we make an Util and avoid the redundancy of these code repeating.

eg:  validateObject(object: any, errorMessage: string) {
   throw new NotFoundException(errorMessage);
}

await this.applicationService.getApplicationByIdAndUser(
        applicationId,
        userToken.userId,
      );

Utils.validateObject(applicationToSubmit,"Student Application not found.");

Reply:
Yes, we can, but I do not see it as part of this PR. Can you please create a tech debit for it to be discussed and implemented where is needed?

Reply:
Agreed. I'll create a technical debt for the same. We can discuss on the scope of the util when we get there.